packages/dds/sequence/src/intervals/sequenceInterval.ts

```diff
@@ -0,0 +1,491 @@
+	 */
+	public getIntervalId(): string {
+		const id = this.properties?.[reservedIntervalIdKey];
+		assert(id !== undefined, 0x5e2 /* interval ID should not be undefined */);
+		return `${id}`;
+	}
+
+	/**
+	 * {@inheritDoc IInterval.union}
+	 * @internal
+	 */
+	public union(b: SequenceInterval) {
+		return new SequenceInterval(
+			this.client,
+			minReferencePosition(this.start, b.start),
+			maxReferencePosition(this.end, b.end),
+			this.intervalType,
+		);
+	}
+
+	/**
+	 * {@inheritDoc ISerializableInterval.addProperties}
+	 * @internal
+	 */
+	public addProperties(
+		newProps: PropertySet,
+		collab: boolean = false,
+		seq?: number,
+		op?: ICombiningOp,
+	): PropertySet | undefined {
+		this.initializeProperties();
+		return this.propertyManager.addProperties(this.properties, newProps, op, seq, collab);
+	}
+
+	/**
+	 * @returns whether this interval overlaps two numerical positions.
+	 */
+	public overlapsPos(bstart: number, bend: number) {
+		const startPos = this.client.localReferencePositionToPosition(this.start);
+		const endPos = this.client.localReferencePositionToPosition(this.end);
+		return endPos > bstart && startPos < bend;
+	}
+
+	/**
+	 * {@inheritDoc IInterval.modify}
+	 * @internal
+	 */
+	public modify(
+		label: string,
+		start: number,
+		end: number,
+		op?: ISequencedDocumentMessage,
+		localSeq?: number,
+		stickiness: IntervalStickiness = IntervalStickiness.END,
+	) {
+		const getRefType = (baseType: ReferenceType): ReferenceType => {
+			let refType = baseType;
+			if (op === undefined) {
+				refType &= ~ReferenceType.SlideOnRemove;
+				refType |= ReferenceType.StayOnRemove;
+			}
+			return refType;
+		};
+
+		let startRef = this.start;
+		if (start !== undefined) {
+			startRef = createPositionReference(
+				this.client,
+				start,
+				getRefType(this.start.refType),
+				op,
+				undefined,
+				localSeq,
+				startReferenceSlidingPreference(stickiness),
+			);
+			if (this.start.properties) {
+				startRef.addProperties(this.start.properties);
+			}
+		}
+
+		let endRef = this.end;
+		if (end !== undefined) {
+			endRef = createPositionReference(
+				this.client,
+				end,
+				getRefType(this.end.refType),
+				op,
+				undefined,
+				localSeq,
+				endReferenceSlidingPreference(stickiness),
+			);
+			if (this.end.properties) {
+				endRef.addProperties(this.end.properties);
+			}
+		}
+
+		const newInterval = new SequenceInterval(this.client, startRef, endRef, this.intervalType);
+		if (this.properties) {
+			newInterval.initializeProperties();
+			this.propertyManager.copyTo(
+				this.properties,
+				newInterval.properties,
+				newInterval.propertyManager,
+			);
+		}
+		return newInterval;
+	}
+
+	private initializeProperties(): void {
+		if (!this.propertyManager) {
+			this.propertyManager = new PropertiesManager();
+		}
+		if (!this.properties) {
+			this.properties = createMap<any>();
+		}
+	}
+}
+
+export function createPositionReferenceFromSegoff(
```

Comment:
Actually `createPositionReferenceFromSegoff` is specific to SequenceInterval, but it is not only internally used in `sequenceInterval.ts`. It is also utilized in `/IntervalIndex/OverlappingSequenceIntervalsIndex.ts`, similarly,  `sequenceIntervalHelpers` is utilized in both two classes. I used to consider moving them to IntervalUtils, but since they are not used in generic intervals, I choose to use the current way to maintain them. I suppose both approaches are okay? waiting for more comments or suggestions about it.

And yes `createPositionReference` is used internally in `sequenceInterval.ts`, I will modify that.